### DISCUSSION TITLE
Convenient entity property dirty handling

### DISCUSSION THREAD

**Discussion by finesoft** [hibernate/hibernate-orm#4926] [Design Proposals]
I want to have such a feature, when the entity is dirty, there is a chance to get the changed entity properties at the last moment before the transaction is committed. It would be better if some relevant annotations can be provided, such as configuring some concerned the name of entity properties and assign these names to a pojo object class (user defined, such as an event object class), the framework can collect changes in the value of these properties and create a new pojo instance and finally trigger these events in conjunction with the CDI event mechanism automatically. Now I am doing these things manually by spi.

**Comment by sebersole** [comment]
First, what the heck is "CDI event mechanism"?

To make sure I understand... You want Hibernate to convert an array into a POJO as part of its internal event handling?  That won't happen, unless I am misunderstanding you.  Converting that array into a POJO has negative performance impacts on quite a few fronts.

**Comment by beikov** [comment]
To make sure I understand you correctly. You are using Hibernate SPIs to implement this right now? If so, then what else do you want Hibernate to provide? Sounds like Hibernate is doing enough already if it allows you to do stuff that you want to achieve. Maybe you should try to explain a bit further what it is that you think is lacking right now and sketch out how you'd like Hibernate to tackle this.

**Comment by yrodiere** [comment]
For what it's worth, Quarkus allows [defining an `Interceptor` as a CDI bean](https://quarkus.io/guides/hibernate-orm#interceptors).

That's definitely not the full story of what you're asking, but I suppose it's a start.

**Comment by gavinking** [comment]
User never responded with clarification of what is being requested here. I’m closing, since I think there are already ways to do approximately this sort of thing.
